### PR TITLE
refactor(multi-env): move getActiveEnv to environmentManager so we can mock in UT

### PR DESCRIPTION
**Pull request by a1exwang**
Global functions in ESM exported by reimporting from another file cannot be mocked by sinon. Take `setActiveEnv` as an example, in core's index.ts, there are code like this:
  
```
export { setActiveEnv } from "./core/middleware/envInfoLoader";
```

This will be compiled to 
```
Object.defineProperty(exports, "setActiveEnv", { enumerable: true, get: function () { return envInfoLoader_1.setActiveEnv; } });
```

From [mozzila docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#description), we can know that this is wrapped as an readonly function (`defineProperty` without `writable: true`). This will cause sinon to fail to mock it.

This is a known issue of sinon and they said these types of issues related to module system is [out of scope](https://github.com/sinonjs/sinon/issues/1711).

One possible solution is to convert the whole core to a common js module which is too much efforts and may affect too many things.

Another solution I have tried is to manually write dependency injection code which will also affect too much because these are util functions used everywhere. And it is not consistent with other parts of CLI.

So I take the third solution to move the function to a modifiable object
- To make is possible to mock with sinon 
- and it also makes sense to make `getActiveEnv` belong to `environmentManager`.

Thanks @tecton to help me debug.

### CODE REVIEW

File: packages/fx-core/src/core/environment.ts [hunk 1]
```diff
@@ -385,6 +392,50 @@ class EnvironmentManager {
       return this.defaultEnvName;
     }
   }
+
+  public getActiveEnv(projectRoot: string): Result<string, FxError> {
+    if (!isMultiEnvEnabled()) {
+      return ok("default");
+    }
+    if (!isValidProject(projectRoot)) {
```
Comment (dooriya) [inline packages/fx-core/src/core/environment.ts:400]: nit: keep return consistent error in envManager if project root path doesn't exist.
Other APIs return `PathNotExistError` while `getActiveEnv` return `InvalidProjectError`:
```ts
if (!(await fs.pathExists(projectPath))) {
  return err(PathNotExistError(projectPath));
}
```

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/src/core/environment.ts [hunk 2]
```diff
@@ -385,6 +392,50 @@ class EnvironmentManager {
...
+    let settingsJson;
+    try {
+      settingsJson = JSON.parse(settingsContent);
+    } catch (error) {
+      return err(
+        InvalidProjectSettingsFileError(
+          `Project settings file is not a valid JSON, error: '${error}'`
+        )
+      );
+    }
```
Comment (qinezh) [inline packages/fx-core/src/core/environment.ts:423]: nit: Here we can use `fs.readJSON` for simplicity.

**Review by qinezh — APPROVED** [review]

**Review by dooriya — APPROVED** [review]

**Review by tecton — APPROVED** [review]

File: packages/fx-core/src/core/environment.ts (hunk 2 above)
Comment (a1exwang) [reply packages/fx-core/src/core/environment.ts:423]: It makes sense. Fixed

File: packages/fx-core/src/core/environment.ts (hunk 1 above)
Comment (a1exwang) [reply packages/fx-core/src/core/environment.ts]: Changed `PathNotExistsError` to be consistent with other APIs.

**Review by nliu-ms — APPROVED** [review]

**Review by chagong — APPROVED** [review]

**Pull request merged**